### PR TITLE
Use 'next/link' for sidenav links

### PR DESCRIPTION
**Pull request by joseph-lozano**
Use Next's `<Link>` component for faster navigation from the side bar. See: https://nextjs.org/docs/app/api-reference/components/link

### CODE REVIEW

File: packages/theme/components/layout/sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -109,7 +115,7 @@ export function Sidebar({pageMap}: SidebarProps) {
             const shouldShowTabs = indexPage.frontMatter?.['show-tabs'] ?? false
             if (shouldShowTabs) {
               return (
-                <NavList.Item key={indexPage.name} href={`${basePath}${indexPage.route}`}>
+                <NavList.Item as={NextLink} key={indexPage.name} href={`${basePath}${indexPage.route}`}>
```
Comment (rezrah) [inline packages/theme/components/layout/sidebar/Sidebar.tsx:118]: Nice. Didn't know NavList Item supported `as`. 

Could you also remove the `basePath` in all instances please, as it should automatically apply now.

Comment (joseph-lozano) [reply packages/theme/components/layout/sidebar/Sidebar.tsx]: Done! I also added `as={NextLink}` to the Breadcrumbs

File: packages/theme/components/layout/index-cards/IndexCards.tsx [hunk 2]
```diff
@@ -31,7 +30,7 @@ export function IndexCards({route, folderData}: IndexCardsProps) {
         if (item.kind !== 'MdxPage') return null
         return (
           <Grid.Column span={{medium: 6}} key={`cell-${item.route}`}>
-            <Card href={`${basePath}${item.route}`} hasBorder style={{width: '100%'}}>
```
Comment (joseph-lozano) [inline packages/theme/components/layout/index-cards/IndexCards.tsx:34]: @rezrah Actually, I think `basePath` should stay here?

File: packages/site/pages/index.mdx [hunk 3]
```diff
@@ -30,7 +30,7 @@ import Link from 'next/link'
 <section style={{['--brand-Card-maxWidth']: '100%',maxWidth: '800px',margin: '0 auto', ['--brand-Grid-spacing-row']: 'var(--brand-Grid-spacing-column-gap)'}}>
   <Grid>
     <Grid.Column span={12}>
-      <Link legacyBehavior href="/getting-started/introduction">
+      <Link legacyBehavior passHref href="/getting-started/introduction">
```
Comment (rezrah) [inline packages/site/pages/index.mdx:33]: Does `passRef` need to be added if legacyBehavior is there? I thought it was one or the other?

**Review by rezrah — APPROVED** [review]
Those links are super fast now. Thanks @joseph-lozano ✨.

Could you add a changeset before you merge please.

One other non-blocking question, otherwise feel free to merge when ready.

File: packages/theme/components/layout/index-cards/IndexCards.tsx (hunk 2 above)
Comment (rezrah) [reply packages/theme/components/layout/index-cards/IndexCards.tsx:34]: Yeah, you'll still need it there. Or you could wrap it in a Next <Link>.

**Pull request merged**